lib/src/usage.dart

```diff
@@ -18,109 +17,107 @@ import 'utils.dart';
-  /// extends past this column will be wrapped at the nearest whitespace (or
-  /// truncated if there is no available whitespace).
+  /// The horizontal character position at which help text is wrapped.
+  ///
+  /// Help that extends past this column will be wrapped at the nearest
+  /// whitespace (or truncated if there is no available whitespace).
   final int? lineLength;
 
-  Usage(this.optionsAndSeparators, {this.lineLength});
+  _Usage(this._optionsAndSeparators, this.lineLength);
 
   /// Generates a string displaying usage information for the defined options.
   /// This is basically the help text shown on the command line.
   String generate() {
-    buffer = StringBuffer();
-
-    calculateColumnWidths();
-
-    for (var optionOrSeparator in optionsAndSeparators) {
+    for (var optionOrSeparator in _optionsAndSeparators) {
       if (optionOrSeparator is String) {
-        // Ensure that there's always a blank line before a separator.
-        if (buffer.isNotEmpty) buffer.write('\n\n');
-        buffer.write(optionOrSeparator);
-        newlinesNeeded = 1;
+        _writeSeparator(optionOrSeparator);
         continue;
       }
-
       var option = optionOrSeparator as Option;
       if (option.hide) continue;
+      _writeOption(option);
+    }
 
-      write(0, getAbbreviation(option));
-      write(1, getLongOption(option));
+    return _buffer.toString();
+  }
 
-      if (option.help != null) write(2, option.help!);
+  void _writeSeparator(String separator) {
+    // Ensure that there's always a blank line before a separator.
+    if (_buffer.isNotEmpty) _buffer.write('\n\n');
+    _buffer.write(separator);
+    _newlinesNeeded = 1;
+  }
 
-      if (option.allowedHelp != null) {
-        var allowedNames = option.allowedHelp!.keys.toList(growable: false);
-        allowedNames.sort();
-        newline();
-        for (var name in allowedNames) {
-          write(1, getAllowedTitle(option, name));
-          write(2, option.allowedHelp![name]!);
-        }
-        newline();
-      } else if (option.allowed != null) {
-        write(2, buildAllowedList(option));
-      } else if (option.isFlag) {
-        if (option.defaultsTo == true) {
-          write(2, '(defaults to on)');
-        }
-      } else if (option.isMultiple) {
-        if (option.defaultsTo != null && option.defaultsTo.isNotEmpty) {
-          write(
-              2,
-              '(defaults to ' +
-                  option.defaultsTo.map((value) => '"$value"').join(', ') +
-                  ')');
-        }
-      } else {
-        if (option.defaultsTo != null) {
-          write(2, '(defaults to "${option.defaultsTo}")');
-        }
+  void _writeOption(Option option) {
+    _write(0, _abbreviation(option));
+    _write(1, _longOption(option));
+
+    if (option.help != null) _write(2, option.help!);
+
+    if (option.allowedHelp != null) {
+      var allowedNames = option.allowedHelp!.keys.toList(growable: false);
```

Comment:
Is the `growable: false` really worth doing here?

Reply:
🤷  it was there before and I didn't want to change it if it _was_ worth doing.

I took a look at it appears it was only added as a precaution when the default changed - it used to be non-growable by default.

https://codereview.chromium.org//12401002


I'll go ahead and remove since it's unlikely to matter and it's noise. (I'm also tempted to use a cascade for the `sort()` call but I guess I'll hold off on that 😝